modules/skybrowser/src/utility.cpp

```diff
@@ -0,0 +1,195 @@
+#include <modules/skybrowser/include/utility.h>
+
+#include <openspace/engine/globals.h>
+#include <openspace/engine/windowdelegate.h>
+#include <openspace/navigation/navigationhandler.h>
+#include <openspace/camera/camera.h>
+#include <glm/gtx/vector_angle.hpp>
+#include <cmath> // For atan2
+#define _USE_MATH_DEFINES
+#include <math.h> // For M_PI
+
+
+namespace openspace::skybrowser {
+
+    // Converts from spherical coordinates in the unit of degrees to cartesian coordianates
+    glm::dvec3 sphericalToCartesian(const glm::dvec2& coords) {
+
+        glm::dvec2 coordsRadians = glm::radians(coords);
+
+        glm::dvec3 cartesian = glm::dvec3(
+            cos(coordsRadians.x) * cos(coordsRadians.y),
+            sin(coordsRadians.x) * cos(coordsRadians.y),
+            sin(coordsRadians.y)
+        );
+
+        return cartesian;
+    }
+
+    // Converts from cartesian coordianates to spherical in the unit of degrees
+    glm::dvec2 cartesianToSpherical(const glm::dvec3& coord) {
+        // Equatorial coordinates RA = right ascension, Dec = declination
+        double ra = atan2(coord.y, coord.x);
+        double dec = atan2(coord.z, glm::sqrt((coord.x * coord.x) + (coord.y * coord.y)));
+
+        ra = ra > 0 ? ra : ra + (2 * M_PI);
```

Comment:
Are you sure this is necessary?  I'd assume that `glm::degrees` would handle the overflow as and *anyway* this line would actually **cause** an overflow since I'd assume it should say:
```
ra = ra < 0 ? ra : ra + (2 * M_PI);
```

Reply:
Yes, it is necessary - it has to do with how wwt reads data. RA must always be positive.